pandas/tests/test_index.py

```diff
@@ -2812,6 +2812,20 @@ def test_view(self):
         result = self._holder(i)
         tm.assert_index_equal(result, i)
 
+class DatetimeAbsoluteLike(DatetimeLike):
+
+    def test_contains(self):
+
+        i = self.create_index()
+
+        self.assertTrue(i[2] in i)
+        self.assertFalse('2012' in i)
+        #  python datetime objects work
+        self.assertTrue(datetime(2013,1,1) in i)
```

Comment:
`Period` doesn't work for `DatetimeIndex`, because it tries to coerce to `Timestamp` and fails:

```
pd.Timestamp(period_index[1])
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-24-22f6327b2caf> in <module>()
----> 1 pd.Timestamp(period_index[1])

pandas/tslib.pyx in pandas.tslib.Timestamp.__new__ (pandas/tslib.c:7638)()

pandas/tslib.pyx in pandas.tslib.convert_to_tsobject (pandas/tslib.c:21357)()

ValueError: Cannot convert Period to Timestamp unambiguously. Use to_timestamp
```